BUG: firedrake-install is now failing with --petsc-int-type=int64

**Describe the bug**
The `firedrake-install` command is now failing with `--petsc-int-type=int64`.

**Error message**
The part of the error message that causes the installation to fail is:

```
tinyasm/matinvert.cpp:7:46: error: cannot convert ‘PetscInt*’ {aka ‘long int*’} to ‘const PetscBLASInt*’ {aka ‘const int*’}
```

--- a/tests/regression/test_star_pc.py
+++ b/tests/regression/test_star_pc.py
@@ -128,7 +128,9 @@ def test_star_equivalence(problem_type, backend):
         a = inner(grad(z), grad(v))*dx + inner(p, q)*dx
 
         bcs = DirichletBC(V.sub(0), Constant((1., 0.)), "on_boundary")
-        nsp = MixedVectorSpaceBasis(V, [V.sub(0), VectorSpaceBasis(constant=True)])
+        nsp = MixedVectorSpaceBasis(
+            V, [V.sub(0), VectorSpaceBasis(constant=True, comm=COMM_WORLD)]
+        )
 
         star_params = {"mat_type": "aij",
                        "snes_type": "ksponly",
@@ -299,7 +301,9 @@ def test_vanka_equivalence(problem_type):
         a = inner(grad(z), grad(v))*dx - inner(p, div(v))*dx - inner(div(z), q)*dx
 
         bcs = DirichletBC(V.sub(0), Constant((1., 0.)), "on_boundary")
-        nsp = MixedVectorSpaceBasis(V, [V.sub(0), VectorSpaceBasis(constant=True)])
+        nsp = MixedVectorSpaceBasis(
+            V, [V.sub(0), VectorSpaceBasis(constant=True, comm=COMM_WORLD)]
+        )
 
         vanka_params = {"mat_type": "aij",
                         "snes_type": "ksponly",

--- a/tinyasm/matinvert.cpp
+++ b/tinyasm/matinvert.cpp
@@ -1,29 +1,19 @@
 #include <petsc.h>
 #include <petscblaslapack.h>
-#include <assert.h>
 
-
-PetscErrorCode mymatinvert(PetscInt* n, PetscScalar* mat, PetscInt* piv, PetscInt* info, PetscScalar* work) {
+PetscErrorCode mymatinvert(PetscInt* n, PetscScalar* mat, PetscBLASInt* piv, PetscInt* info, PetscScalar* work) {
     // Declare BLAS compatible variables
     PetscBLASInt n_blas, info_blas;
-    PetscBLASInt *piv_blas;
 
     // Convert function arguments to BLAS compatible ints
     PetscCall(PetscBLASIntCast(*n, &n_blas));
-    // piv is OUT for getrf and IN for getri so we don't need to check
-    // whether the cast is safe
-    piv_blas = (PetscBLASInt*) piv;
     PetscCall(PetscBLASIntCast(*info, &info_blas));
 
-    assert(*n == n_blas);
+    PetscCallBLAS("LAPACKgetrf", LAPACKgetrf_(&n_blas, &n_blas, mat, &n_blas, piv, &info_blas));
+    PetscCallBLAS("LAPACKgetri", LAPACKgetri_(&n_blas, mat, &n_blas, piv, work, &n_blas, &info_blas));
 
-    //~ printf("n %" PetscInt_FMT " \n", *n);
-    //~ printf("n_blas %" PetscBLASInt_FMT " \n", n_blas);
-    //~ printf("piv %" PetscInt_FMT " \n", *piv);
-    //~ printf("piv_blas %" PetscBLASInt_FMT " \n", piv_blas);
-
-    PetscCallBLAS("LAPACKgetrf",LAPACKgetrf_(&n_blas, &n_blas, mat, &n_blas, piv_blas, &info_blas));
-    PetscCallBLAS("LAPACKgetri",LAPACKgetri_(&n_blas, mat, &n_blas, piv_blas, work, &n_blas, &info_blas));
+    // Cast info back since it has intent out
+    *info = (PetscInt) info_blas;
     return 0;
 }
 

--- a/tinyasm/matinvert.cpp
+++ b/tinyasm/matinvert.cpp
@@ -1,19 +1,11 @@
 #include <petsc.h>
 #include <petscblaslapack.h>
 
-PetscErrorCode mymatinvert(PetscInt* n, PetscScalar* mat, PetscBLASInt* piv, PetscInt* info, PetscScalar* work) {
-    // Declare BLAS compatible variables
-    PetscBLASInt n_blas, info_blas;
-
-    // Convert function arguments to BLAS compatible ints
-    PetscCall(PetscBLASIntCast(*n, &n_blas));
-    PetscCall(PetscBLASIntCast(*info, &info_blas));
-
-    PetscCallBLAS("LAPACKgetrf", LAPACKgetrf_(&n_blas, &n_blas, mat, &n_blas, piv, &info_blas));
-    PetscCallBLAS("LAPACKgetri", LAPACKgetri_(&n_blas, mat, &n_blas, piv, work, &n_blas, &info_blas));
-
-    // Cast info back since it has intent out
-    *info = (PetscInt) info_blas;
-    return 0;
+PetscErrorCode mymatinvert(PetscBLASInt* n, PetscScalar* mat, PetscBLASInt* piv, PetscBLASInt* info, PetscScalar* work) {
+    PetscCallBLAS("LAPACKgetrf", LAPACKgetrf_(n, n, mat, n, piv, info));
+    PetscCheck(!(*info), PETSC_COMM_SELF, PETSC_ERR_LIB, "TinyASM error calling ?getrf in mymatinvert");
+    PetscCallBLAS("LAPACKgetri", LAPACKgetri_(n, mat, n, piv, work, n, info));
+    PetscCheck(!(*info), PETSC_COMM_SELF, PETSC_ERR_LIB, "TinyASM error calling ?getri in mymatinvert");
+    return PETSC_SUCCESS;
 }
 

--- a/tinyasm/matinvert.cpp
+++ b/tinyasm/matinvert.cpp
@@ -3,8 +3,12 @@
 
 
 PetscErrorCode mymatinvert(PetscInt* n, PetscScalar* mat, PetscInt* piv, PetscInt* info, PetscScalar* work) {
-    PetscCallBLAS("LAPACKgetrf",LAPACKgetrf_(n,n,mat,n,piv,info));
-    PetscCallBLAS("LAPACKgetri",LAPACKgetri_(n,mat, n, piv,work,n,info));
+    PetscBLASInt *n_blas, *piv_blas, *info_blas;
+    PetscCall(PetscBLASIntCast(*n, n_blas));
+    PetscCall(PetscBLASIntCast(*piv, piv_blas));
+    PetscCall(PetscBLASIntCast(*info, info_blas));
+    PetscCallBLAS("LAPACKgetrf",LAPACKgetrf_(n_blas, n_blas, mat, n_blas, piv_blas, info_blas));
+    PetscCallBLAS("LAPACKgetri",LAPACKgetri_(n_blas, mat, n_blas, piv_blas, work, n_blas, info_blas));
     return 0;
 }
 

--- a/tinyasm/matinvert.cpp
+++ b/tinyasm/matinvert.cpp
@@ -3,12 +3,12 @@
 
 
 PetscErrorCode mymatinvert(PetscInt* n, PetscScalar* mat, PetscInt* piv, PetscInt* info, PetscScalar* work) {
-    PetscBLASInt *n_blas, *piv_blas, *info_blas;
-    PetscCall(PetscBLASIntCast(*n, n_blas));
-    PetscCall(PetscBLASIntCast(*piv, piv_blas));
-    PetscCall(PetscBLASIntCast(*info, info_blas));
-    PetscCallBLAS("LAPACKgetrf",LAPACKgetrf_(n_blas, n_blas, mat, n_blas, piv_blas, info_blas));
-    PetscCallBLAS("LAPACKgetri",LAPACKgetri_(n_blas, mat, n_blas, piv_blas, work, n_blas, info_blas));
+    PetscBLASInt n_blas, piv_blas, info_blas;
+    PetscCall(PetscBLASIntCast(*n, &n_blas));
+    PetscCall(PetscBLASIntCast(*piv, &piv_blas));
+    PetscCall(PetscBLASIntCast(*info, &info_blas));
+    PetscCallBLAS("LAPACKgetrf",LAPACKgetrf_(&n_blas, &n_blas, mat, &n_blas, &piv_blas, &info_blas));
+    PetscCallBLAS("LAPACKgetri",LAPACKgetri_(&n_blas, mat, &n_blas, &piv_blas, work, &n_blas, &info_blas));
     return 0;
 }
 

--- a/tinyasm/matinvert.cpp
+++ b/tinyasm/matinvert.cpp
@@ -1,14 +1,29 @@
 #include <petsc.h>
 #include <petscblaslapack.h>
+#include <assert.h>
 
 
 PetscErrorCode mymatinvert(PetscInt* n, PetscScalar* mat, PetscInt* piv, PetscInt* info, PetscScalar* work) {
-    PetscBLASInt n_blas, piv_blas, info_blas;
+    // Declare BLAS compatible variables
+    PetscBLASInt n_blas, info_blas;
+    PetscBLASInt *piv_blas;
+
+    // Convert function arguments to BLAS compatible ints
     PetscCall(PetscBLASIntCast(*n, &n_blas));
-    PetscCall(PetscBLASIntCast(*piv, &piv_blas));
+    // piv is OUT for getrf and IN for getri so we don't need to check
+    // whether the cast is safe
+    piv_blas = (PetscBLASInt*) piv;
     PetscCall(PetscBLASIntCast(*info, &info_blas));
-    PetscCallBLAS("LAPACKgetrf",LAPACKgetrf_(&n_blas, &n_blas, mat, &n_blas, &piv_blas, &info_blas));
-    PetscCallBLAS("LAPACKgetri",LAPACKgetri_(&n_blas, mat, &n_blas, &piv_blas, work, &n_blas, &info_blas));
+
+    assert(*n == n_blas);
+
+    //~ printf("n %" PetscInt_FMT " \n", *n);
+    //~ printf("n_blas %" PetscBLASInt_FMT " \n", n_blas);
+    //~ printf("piv %" PetscInt_FMT " \n", *piv);
+    //~ printf("piv_blas %" PetscBLASInt_FMT " \n", piv_blas);
+
+    PetscCallBLAS("LAPACKgetrf",LAPACKgetrf_(&n_blas, &n_blas, mat, &n_blas, piv_blas, &info_blas));
+    PetscCallBLAS("LAPACKgetri",LAPACKgetri_(&n_blas, mat, &n_blas, piv_blas, work, &n_blas, &info_blas));
     return 0;
 }
 

--- a/tinyasm/tinyasm.cpp
+++ b/tinyasm/tinyasm.cpp
@@ -49,7 +49,7 @@ class BlockJacobi {
 
                 int numBlocks = dofsPerBlock.size();
                 PetscInt dof;
-                int biggestBlock = 0;
+                PetscInt biggestBlock = 0;
                 for(int p=0; p<numBlocks; p++) {
                     dof = dofsPerBlock[p].size();
                     biggestBlock = max(biggestBlock, dof);
@@ -108,9 +108,10 @@ class BlockJacobi {
 
 
         PetscInt solve(const PetscScalar* __restrict b, PetscScalar* __restrict x) {
-            PetscInt dof, ierr;
+            PetscBLASInt dof;
+            PetscInt ierr;
             PetscScalar dOne = 1.0;
-            PetscInt one = 1;
+            PetscBLASInt one = 1;
             PetscScalar dZero = 0.0;
 
             const PetscScalar *matvalues;
@@ -324,9 +325,9 @@ PetscErrorCode PCView_TinyASM(PC pc, PetscViewer viewer) {
         PetscInt biggestblock = *std::max_element(blocksizes.begin(), blocksizes.end());
         PetscScalar avgblock = std::accumulate(blocksizes.begin(), blocksizes.end(), 0.)/nblocks;
         ierr = PetscViewerASCIIPushTab(viewer);CHKERRQ(ierr);
-        ierr = PetscViewerASCIIPrintf(viewer, "TinyASM (block Jacobi) preconditioner with %d blocks\n", nblocks);CHKERRQ(ierr);
+        ierr = PetscViewerASCIIPrintf(viewer, "TinyASM (block Jacobi) preconditioner with %" PetscInt_FMT " blocks\n", nblocks);CHKERRQ(ierr);
         ierr = PetscViewerASCIIPrintf(viewer, "Average block size %f \n", avgblock);CHKERRQ(ierr);
-        ierr = PetscViewerASCIIPrintf(viewer, "Largest block size %d \n", biggestblock);CHKERRQ(ierr);
+        ierr = PetscViewerASCIIPrintf(viewer, "Largest block size %" PetscInt_FMT " \n", biggestblock);CHKERRQ(ierr);
         ierr = PetscViewerASCIIPopTab(viewer);CHKERRQ(ierr);
     }
     return 0;
@@ -428,6 +429,6 @@ PYBIND11_MODULE(_tinyasm, m) {
               auto blockjacobi = new BlockJacobi(dofsPerBlock, globalDofsPerBlock, localsize, newsf);
               pc->data = (void*)blockjacobi;
               ierr = PetscLogEventEnd(PC_tinyasm_SetASMLocalSubdomains, pc, 0, 0, 0);CHKERRQ(ierr);
-              return ierr;
+              return (int) ierr;
           });
 }

--- a/tinyasm/tinyasm.cpp
+++ b/tinyasm/tinyasm.cpp
@@ -26,7 +26,7 @@ namespace py = pybind11;
 
 PetscLogEvent PC_tinyasm_SetASMLocalSubdomains, PC_tinyasm_apply, PC_tinyasm_setup;
 
-PetscErrorCode mymatinvert(PetscInt* n, PetscScalar* mat, PetscInt* piv, PetscInt* info, PetscScalar* work);
+PetscErrorCode mymatinvert(PetscInt* n, PetscScalar* mat, PetscBLASInt* piv, PetscInt* info, PetscScalar* work);
 
 class BlockJacobi {
     public:
@@ -41,7 +41,7 @@ class BlockJacobi {
         Mat *localmats;
 
         vector<IS> dofis;
-        vector<PetscInt> piv;
+        vector<PetscBLASInt> piv;
         vector<PetscScalar> fwork;
 
         BlockJacobi(vector<vector<PetscInt>> _dofsPerBlock, vector<vector<PetscInt>> _globalDofsPerBlock, int localSize, PetscSF _sf)
@@ -58,7 +58,7 @@ class BlockJacobi {
                 workb = vector<PetscScalar>(biggestBlock, 0);
                 localb = vector<PetscScalar>(localSize, 0);
                 localx = vector<PetscScalar>(localSize, 0);
-                piv = vector<PetscInt>(biggestBlock, 0.);
+                piv = vector<PetscBLASInt>(biggestBlock, 0.);
                 iota(piv.begin(), piv.end(), 1);
                 fwork = vector<PetscScalar>(biggestBlock, 0.);
                 localmats_aij = NULL;

--- a/tinyasm/tinyasm.cpp
+++ b/tinyasm/tinyasm.cpp
@@ -26,7 +26,7 @@ namespace py = pybind11;
 
 PetscLogEvent PC_tinyasm_SetASMLocalSubdomains, PC_tinyasm_apply, PC_tinyasm_setup;
 
-PetscErrorCode mymatinvert(PetscInt* n, PetscScalar* mat, PetscBLASInt* piv, PetscInt* info, PetscScalar* work);
+PetscErrorCode mymatinvert(PetscBLASInt* n, PetscScalar* mat, PetscBLASInt* piv, PetscBLASInt* info, PetscScalar* work);
 
 class BlockJacobi {
     public:
@@ -91,14 +91,14 @@ class BlockJacobi {
         }
 
         PetscInt updateValuesPerBlock(Mat P) {
-            PetscInt ierr, dof;
+            PetscInt ierr;
+            PetscBLASInt dof, info;
             int numBlocks = dofsPerBlock.size();
             ierr = MatCreateSubMatrices(P, numBlocks, dofis.data(), dofis.data(), localmats_aij ? MAT_REUSE_MATRIX : MAT_INITIAL_MATRIX, &localmats_aij);CHKERRQ(ierr);
-            PetscInt info;
             PetscScalar *vv;
             for(int p=0; p<numBlocks; p++) {
                 ierr = MatConvert(localmats_aij[p], MATDENSE, localmats[p] ? MAT_REUSE_MATRIX : MAT_INITIAL_MATRIX,&localmats[p]);CHKERRQ(ierr);
-                PetscInt dof = dofsPerBlock[p].size();
+                PetscCall(PetscBLASIntCast(dofsPerBlock[p].size(), &dof));
                 ierr = MatDenseGetArrayWrite(localmats[p],&vv);CHKERRQ(ierr);
                 if (dof) mymatinvert(&dof, vv, piv.data(), &info, fwork.data());
                 ierr = MatDenseRestoreArrayWrite(localmats[p],&vv);CHKERRQ(ierr);
@@ -108,10 +108,9 @@ class BlockJacobi {
 
 
         PetscInt solve(const PetscScalar* __restrict b, PetscScalar* __restrict x) {
-            PetscBLASInt dof;
             PetscInt ierr;
             PetscScalar dOne = 1.0;
-            PetscBLASInt one = 1;
+            PetscBLASInt dof, one = 1;
             PetscScalar dZero = 0.0;
 
             const PetscScalar *matvalues;

